src/test/java/rest/SearchResourceTest.java

```diff
@@ -274,6 +294,140 @@ public void testGetPersonByFullName()
+                .extract().body()
+                .jsonPath().getList(".", PersonDTO_OUT.class); //https://stackoverflow.com/a/53006523
+
+        //Assert
+        assertThat((result), equalTo(expResult));
+    }
+    
+    @Test
+    public void testGetPersonsByCity_Exception1(){
+        //Arrange
+        String zip = ""; //bad input
+        String city = ""; 
+        
+        given()
+                .get("search/city?zip={zip}&city={city}", zip, city).then()
+                .assertThat()
+                .statusCode(HttpStatus.BAD_REQUEST_400.getStatusCode()).
+                body("code", equalTo(400)).
+                body("message", equalTo("Missing Input"));
+    }
+    
+    @Test
+    public void testGetPersonsByCity_Exception2(){
+        //Arrange
+        String zip = "123"; //Incorrect input
+        String city = "NoCity"; 
+        
+        given()
+                .get("search/city?zip={zip}&city={city}", zip, city).then()
+                .assertThat()
+                .statusCode(HttpStatus.NOT_FOUND_404.getStatusCode()).
+                body("code", equalTo(404)).
+                body("message", equalTo("No Persons lives in that city."));
+    }
+    
+    @Test 
+    public void testGetAllZipCodes(){
+        //Arrange
+        List<Integer> expResult = new ArrayList();
+        expResult.add(Integer.valueOf(city1.getZipCode())); //2800
+        expResult.add(Integer.valueOf(city2.getZipCode())); //8000
+        expResult.add(Integer.valueOf(city3.getZipCode())); //9900
+        List<Integer> result;
+        
+        //Act
+        result = get("/search/zip/").then()
+                .assertThat()
+                .statusCode(HttpStatus.OK_200.getStatusCode())
+                .extract().body()
+                .jsonPath().getList("", int.class); //could probably be Integer.class but I'm scared to change anything now that it works.
+
+        //Assert
+////        System.out.println(result.getClass());
+////        System.out.println(expResult.getClass());
+////        System.out.println(expResult.get(0).getClass());
+////        System.out.println(result.get(0).getClass());
+////        System.out.println(result);
+////        System.out.println(expResult);
+        //assertThat((result), equalTo(expResult));
+        //assertEquals(expResult, result);
+        //assertThat(expResult, Matchers.containsInAnyOrder(result.get(0), result.get(1), result.get(2)));
+        assertThat("ARE THEY EQUAL", result, containsInAnyOrder(expResult.toArray()));
+    }
+    
+    /**
+     * Simulates an empty db
+     */
+    @Tag("skipBeforeEach")
+    @Test public void testGetAllZipCodes_Exception1(){
+        //Assert
+        given().
+                get("/search/zip/").then().
+                statusCode(HttpStatus.BAD_REQUEST_400.getStatusCode()).
+                assertThat().
+                body("code", equalTo(400)).
+                body("message", equalTo("No cities in the database."));
+    }
```

Comment:
Could you explain how this works ? :)

Reply:
I documented it on [line 150](https://github.com/MalteMagnussen/CA2/pull/53/files/d80fc10e8b9a9c20a4427408cf88dd59a6ea6aed#diff-a138706ed1fa2327b11739b17e647eb7R150) using the following stackoverflow link:
https://stackoverflow.com/a/49694288

Essentially we, with JUnit 5, can take use of `TestInfo`, an element that contains information about the test being run.

In our @BeforeEach (see above) I made some changes to take into account if the test is tagget with `skipBeforeEach` (line 420).

I essentially skip the @BeforeEach for the method `testGetAllZipCodes_Exception1()` because I need to test exception throwing for an empty database.

Reply:
Thanks, great explanation.